controller/node_controller.go

```diff
@@ -403,6 +407,32 @@ func (nc *NodeController) syncNode(key string) (err error) {
 			return err
 		}
 	} else {
+		// check if the role of the node got changed
+		kubeNodeLabels := kubeNode.GetLabels()
+		if len(kubeNodeLabels) != 0 {
+			isCurrentNodeWorker := false
```

Comment:
This is not `CurrentNode`. Node controller check for all nodes since if the node is lost, the manager won't be running.